sci-libs/scotch/scotch-6.0.9.recipe

```diff
@@ -0,0 +1,226 @@
+	cmd:mtst$commandSuffix = $portVersionCompat
+	"
+REQUIRES_tools="
+	scotch$secondaryArchSuffix == $portVersion base
+	haiku$secondaryArchSuffix
+	lib:liblzma$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:liblzma$secondaryArchSuffix
+	devel:libbz2$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:bison
+	cmd:gcc$secondaryArchSuffix
+	cmd:flex
+	cmd:make
+	"
+
+TEST_REQUIRES="
+	cmd:diff
+	"
+
+defineDebugInfoPackage scotch$secondaryArchSuffix \
+	"$libDir"/libesmumps.so \
+	"$libDir"/libscotch.so \
+	"$libDir"/libscotcherr.so \
+	"$libDir"/libscotcherrexit.so\
+	"$libDir"/libscotchmetis.so
+
+PATCH()
+{
+	# make install will fail because these binaries are missing
+	# This could be caused by the ptscotch target which has been disabled on this release
+	# Thus, this is not going to be a permanent patch, hence it is not
+	# in the main patch set
+	sed -i 's|-\$(CP) -f \.\./bin/d\[agm\]\*\$(EXE) \$(bindir)||g' src/Makefile
+}
+
+BUILD()
+{
+	export LIBRARY_PATH=$LIBRARY_PATH:$sourceDir/lib
+
+	cp "$portDir"/additional-files/Makefile.inc."${effectiveTargetArchitecture}"_pc_haiku.shlib src/Makefile.inc
+
+	make -C src $jobArgs scotch esmumps
+}
+
+
+INSTALL()
+{
+	declare -a destdirs=("$commandBinDir" "$includeDir" "$includeDir" "$libDir" "$libDir"/pkgconfig "$dataDir" "$manDir")
+	for i in "${destdirs[@]}"
+	do
+		mkdir -p ${i}
+	done
+
+	make -C src $jobArgs bindir=$commandBinDir includedir=$includeDir libdir=$libDir datarootdir=$dataDir mandir=$manDir install
+
+	# create symlinks
+	declare -a libraries=("libesmumps" "libscotch" "libscotcherr" "libscotcherrexit" "libscotchmetis")
+	for soname in "${libraries[@]}"
+	do
+		ln -s ${libDir}/${soname}.so ${libDir}/${soname}.so.$libVersion
+		ln -s ${libDir}/${soname}.so ${libDir}/${soname}.so.${libVersion%%.*}
+	done
+
+	cat <<-EOF > $libDir/pkgconfig/scotch.pc
+		prefix=${prefix}
+		exec_prefix=${prefix}
+		libdir=\${prefix}/develop/lib
+		includedir=\${prefix}/develop/headers
+		Name: scotch
+		Description: ${SUMMARY}
+		Version: ${portVersion}
+		URL: ${HOMEPAGE}
+		Libs: -L\${libdir} -lscotcherr -lscotch
+		Private: -lz -llzma -lbz2 -lm
+		Cflags: -I\${includedir}
+	EOF
+
+	cat <<-EOF > $libDir/pkgconfig/scotchmetis.pc
+		prefix=${prefix}
+		libdir=\${prefix}/develop/lib
+		includedir=\${prefix}/develop/headers
+		Name: scotchmetis
+		Description: ${SUMMARY}
+		Version: ${portVersion}
+		URL: ${HOMEPAGE}
+		Libs: -L\${libdir} -lscotchmetis -lscotcherr -lscotch
+		Private: -lz -llzma -lbz2 -lm
+		Cflags: -I\${includedir}
+	EOF
+
+	prepareInstalledDevelLibs `printf '%s\n' "$(IFS=' '; printf '%s' "${libraries[*]}")"`
+	fixPkgconfig
+
+	# copy pdf documentation
+	mkdir -p $developDocDir
+	cp -rd doc/*.pdf $developDocDir
+
+	# devel package
+	packageEntries devel $developDir
+
+	# tools package
+	packageEntries tools $commandBinDir $manDir
+}
+
+TEST()
+{
+
```

Comment:
With this in place, the tests run OK here for gcc2h (secondary architecture), see Terminal output in the comment for this PR:
```diff
+	export LIBRARY_PATH=$LIBRARY_PATH:$sourceDir/lib
+
```

Reply:
x86_64 is ok too. I see no glitches, now.

FYI tests were hanging due to a bug in test suite (a path in tgt file used to perform the diffs). Incidentally, adding LIBRARY_PATH solves this as well.

I think it could be merged, if it's ok from your side.

Reply:
@extrowerk could you give it a second look also?